drivers/gps/nrf9160_gps/nrf9160_gps.c

```diff
@@ -763,7 +763,30 @@ static int start(const struct device *dev, struct gps_config *cfg)
 		return -EIO;
 	}
 
-	err = nrf_modem_gnss_use_case_set(drv_data->config.use_case);
+	uint8_t use_case = drv_data->config.use_case;
+
+#ifdef CONFIG_NRF_CLOUD_AGPS_FILTERED
+	/* prevent modem from wasting time downloading missing ephemerides */
+	use_case |= NRF_MODEM_GNSS_USE_CASE_SCHED_DOWNLOAD_DISABLE;
+
+	err = nrf_modem_gnss_elevation_threshold_set(CONFIG_NRF_CLOUD_AGPS_ELEVATION_MASK);
+	if (err) {
+		LOG_ERR("Failed to set elevation threshold to %d: %d",
+			CONFIG_NRF_CLOUD_AGPS_ELEVATION_MASK, err);
+		return err;
+	}
+#endif
+
+	err = nrf_modem_gnss_use_case_set(use_case);
+#ifdef CONFIG_NRF_CLOUD_AGPS_FILTERED
+	if (err) {
+		LOG_WRN("Could not disable scheduled GNSS downloads: %d", err);
+		use_case = NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START;
```

Comment:
should this be `|=` ?

Reply:
No.  Disabling scheduled downloads is only supported in mfw_1.3.1 or greater.  So if we fail to set it, this tries again without it.  It would be better though to mask off `NRF_MODEM_GNSS_USE_CASE_SCHED_DOWNLOAD_DISABLE` instead.

Reply:
ah sorry i meant `use_case &= ~NRF_MODEM_GNSS_USE_CASE_SCHED_DOWNLOAD_DISABLE;` like is done below, since the  use_case comes from `drv_data->config.use_case`

Reply:
done